src/utils/clickHandler.js

```diff
@@ -92,14 +94,17 @@ module.exports = function clickHandler (e) {
         }
     }
 
-    self.executeI13nEvent('click', {i13nNode: self.getI13nNode(), e:e}, function clickBeaconCallback () {
-        if (isRedirectLink) {
-            if (isFormSubmit(target)) {
-                // if the button has no form linked, then do nothing
-                target.form && target.form.submit();
-            } else {
-                document.location.assign(href);
+    // set-immediate to get click event fired with next tick, in case we need to update something before sending beacons
+    setImmediate(function executeI13nEvent() {
```

Comment:
I'm actually not sure you want to do this. Imagine the toggle button scenario:

1) click on button in "off" state
2) click handler is invoked and setImmediate is called
3) component calls setState on itself, triggering a re-render
4) button renders in "on" state
5) this code finally executes with the click beacon now in the "on" state, but we wanted to send it in the "off" state.

But I may be wrong...Can you confirm that we won't send the wrong beacon in this case?